### ISSUE TITLE
Training stability of mix ln

### ISSUE THREAD

**Issue by x54-729** [pixeli99/MixLN#1]
I'm trying to train a 7B model with mix ln now, but I noticed that the training loss is not that stable:
![image](https://github.com/user-attachments/assets/894ca30d-60d3-4449-b694-6042be39a6dc)
Gradient norm is not stable either:
![image](https://github.com/user-attachments/assets/76eeb24b-03cd-47bb-b193-7fbba91d830b)
I wonder if there are some experiments about the training stability of mix layer norm?

**Comment by pixeli99** [comment]
Hi, @x54-729 ,
Thank you for your attention. If it's convenient, please paste your hyperparameters, such as lr, number of post-norm layers, etc.

**Comment by pixeli99** [comment]
In our experiments, for the 7B model, we used a learning rate of 5e-4 with 2 post-normalization layers.

**Comment by x54-729** [comment]
hidden_size=4096
nun_layers=51
num_heads=32
num_kv_heads=8
post-norm layers=3
lr=6e-4 with 2000 steps warmup

**Comment by pixeli99** [comment]
It seems you need more warmup steps, for example, 10% of the total training steps. In the reported results, we used 1w steps for warmup.
